cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -295,6 +298,7 @@ func (csr *ClusterStateRegistry) UpdateNodes(nodes []*apiv1.Node, nodeInfosForGr
 	if err != nil {
 		return err
 	}
+	cloudProviderNodesRemoved := csr.getCloudProviderDeletedNodes(nodes, cloudProviderNodeInstances)
```

Comment:
This was also attempting to match with an existing interface design, but could be consolidated for simplicity.  The other consideration used to be to avoid locking the ClusterStateRegistry for too long, which might be the reason for the separation of a discovery phase and updating the cluster state.  It shouldn't be an issue to combine these for this use case.